content/kb/tutorials/databases/public-gsheet.md

```diff
@@ -47,41 +47,27 @@ As the `secrets.toml` file above is not committed to GitHub, you need to pass it
 
 ![Secrets manager screenshot](/images/databases/edit-secrets.png)
 
-## Add gsheetsdb to your requirements file
-
-Add the [gsheetsdb](https://github.com/betodealmeida/gsheets-db-api) package to your `requirements.txt` file, preferably pinning its version (replace `x.x.x` with the version you want installed):
-
-```bash
-# requirements.txt
-gsheetsdb==x.x.x
-```
-
 ## Write your Streamlit app
 
 Copy the code below to your Streamlit app and run it.
 
 ```python
 # streamlit_app.py
 
+import pandas as pd
 import streamlit as st
-from gsheetsdb import connect
-
-# Create a connection object.
-conn = connect()
 
-# Perform SQL query on the Google Sheet.
+# Read in data from the Google Sheet.
 # Uses st.cache_data to only rerun when the query changes or after 10 min.
 @st.cache_data(ttl=600)
-def run_query(query):
-    rows = conn.execute(query, headers=1)
-    rows = rows.fetchall()
-    return rows
+def load_data(sheets_url):
+    csv_url = sheets_url.replace("/edit#gid=", "/export?format=csv&gid=")
+    return pd.read_csv(csv_url, index_col=0)
 
-sheet_url = st.secrets["public_gsheets_url"]
-rows = run_query(f'SELECT * FROM "{sheet_url}"')
+df = load_data(st.secrets["public_gsheets_url"])
 
 # Print results.
-for row in rows:
+for _, row in df.iterrows():
```

Comment:
Could use itertuples instead of iterrows, which simplifies the code and is faster.

Reply:
Good call, @blackary! I've used itertuples instead in 0024286.